tools/_setup_generation/step_gallery.py

```diff
@@ -0,0 +1,119 @@
+# - type: The type of the item (code, video or article)
+# - data-keywords: A comma separated list of keywords
+# - short_description: A short description of the item
+#
+# ##########################################################################################
+import os
+import re
+from typing import List
+
+from .setup import SetupStep, Setup
+from .items import FolderItem, FileItem, Item
+from .items.exceptions import WrongHeader, NoHeader, NoIndexFile
+
+
+class GalleryStep(SetupStep):
+
+    # The name of the gallery folder
+    GALLERY_FOLDER_NAME = "gallery"
+
+    # The placeholders in the gallery index file to be replaced by the generated content
+    GALLERY_PLACEHOLDER = r"\[LIST_FOR_GALLERY\]"
+
+    def __init__(self):
+        self.GALLERY_FOLDER_PATHS = None
+        self.GALLERY_INDEX_PATH = None
+        self.GALLERY_INDEX_TPL_PATH = None
+        self.gallery = ""
+
+
+    def enter(self, setup: Setup):
+        self.GALLERY_FOLDER_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME)
+        self.GALLERY_INDEX_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME, "index.md")
+        self.GALLERY_INDEX_TEMPLATE_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME, "index.md_template")
+  
+    def get_id(self) -> str:
+        return "gallery"
+
+    def get_description(self) -> str:
+        return "Generating the list of items on the gallery page."
+
+    def setup(self, setup: Setup) -> None:
+        try:
+            # Parse the gallery folders and get all the items
+            gallery_items = self._get_list_of_items(self.GALLERY_FOLDER_PATH)
+
+            # Generate content for the kb index page
+            self.gallery = self._build_gallery_content(gallery_items)
+            self._update_gallery_index_file()
+
+            print(f"{len(gallery_items)} gallery processed.")
+        except Exception as e:
+            print(f"WARNING - Exception raised while generating the gallery index:\n{e}")
+
+    def exit(self, setup: Setup):
+        pass
+
+    @staticmethod
+    def _get_list_of_items(folder_path) -> List[Item]:
+        """List all folders in folder_path and return a list of Item objects built from the sub folders."""
+        items = []
+        for sub_folder in os.listdir(folder_path):
+            path = os.path.join(folder_path, sub_folder)
+            isdir = os.path.isdir(path)
+            try:
+                if isdir:
+                    if sub_folder != "images":
+                        try:
+                            item = FolderItem(folder_path, sub_folder)
+                            items.append(item)
+                        except NoIndexFile as e:
+                            print(f"WARNING - ", e)
+                else:
+                    if not sub_folder.startswith("index.md"):
+                        item = FileItem(folder_path, sub_folder)
+                        items.append(item)
+            except NoHeader as e:
```

Comment:
```suggestion
            except (NoHeader, WrongHeader) as e:
```
don't repeat the code